agr_literature_service/lit_processing/dump_json_data.py

```diff
@@ -0,0 +1,576 @@
+
+    reference_id_to_authors = get_author_data(db_session, mod_id)
+
+    log.info("Getting comment/correction data from the database...")
+
+    reference_id_to_comment_correction_data = get_comment_correction_data(db_session,
+                                                                          reference_id_to_pmid)
+
+    mod_id_to_mod = dict([(x.mod_id, x.abbreviation) for x in db_session.query(ModModel).all()])
+
+    log.info("Getting mod reference type data from the database...")
+
+    reference_id_to_mod_reference_types = get_mod_reference_type_data(db_session)
+
+    log.info("Getting mod_corpus_association data from the database...")
+    reference_id_to_mod_corpus_data = get_mod_corpus_association_data(db_session,
+                                                                      mod_id_to_mod)
+
+    log.info("Getting journal data from the database...")
+
+    resource_id_to_journal = get_journal_data(db_session)
+
+    log.info("Getting mesh term data from the database...")
+
+    reference_id_to_mesh_terms = get_mesh_term_data(db_session, mod_id)
+
+    db_session.close()
+
+    log.info("Getting data from Reference table and generating json file...")
+    try:
+        get_reference_data_and_generate_json(mod_id, mod, reference_id_to_source,
+                                             reference_id_to_xrefs,
+                                             reference_id_to_authors,
+                                             reference_id_to_comment_correction_data,
+                                             reference_id_to_mod_reference_types,
+                                             reference_id_to_mesh_terms,
+                                             reference_id_to_mod_corpus_data,
+                                             resource_id_to_journal,
+                                             json_path + json_file, datestamp)
+    except Exception as e:
+        print("Error occurred when retrieving data from Reference data and generating json file: " + str(e))
+        return
+
+    log.info("Uploading json file to s3...")
+    try:
+        upload_json_file_to_s3(json_path, json_file, datestamp, ondemand)
+    except Exception as e:
+        print("Error occurred when uploading json file to s3: " + str(e))
+        return
+
+    log.info("DONE!")
+
+
+def generate_json_file(metaData, data, filename_with_path):
+
+    try:
+        jsonData = {"data": data,
+                    "metaData": metaData}
+        fw = open(filename_with_path, 'w')
+        fw.write(json.dumps(jsonData, indent=4, sort_keys=True))
+        fw.close()
+    except Exception as e:
+        print("Error when generating", filename_with_path, ": " + str(e))
+        exit()
+
+
+def upload_json_file_to_s3(json_path, json_file, datestamp, ondemand):
+
+    env_state = environ.get('ENV_STATE', 'develop')
+    if env_state == 'build':
+        env_state = 'develop'
+    if env_state == 'test':
+        return
```

Comment:
Here's where I meant in the other comment that env_state stays ENV_STATE instead of preventing upload

Reply:
If ENV_STATE is set to 'test', the script won't upload the files to s3; if ENV_STATE is not set or set to 'build', then upload the files 'develop'; otherwise upload the files to 'prod'. So you think when ENV_STATE is set to 'test', the script will still upload the files to 'develop'?

Reply:
I agree! We should not upload the files to s3 while testing script.  I think the current code will not, but I will double check

Reply:
Oh, I am sorry, you're right, it doesn't upload.